ee/billing/quota_limiting.py

```diff
@@ -231,7 +241,27 @@ def org_quota_limited_until(
             )
         return None
 
-    # 1b. never drop
+    # 1b. AI credits have no grace period and ignore never_drop_data - immediately limit
+    if resource == QuotaResource.AI_CREDITS_USED:
+        report_organization_action(
+            organization,
+            "org_quota_limited_until",
+            properties={
+                "event": "suspended",
+                "current_usage": usage + todays_usage,
+                "resource": resource.value,
+                "trust_score": trust_score,
+            },
+        )
+        update_organization_usage_fields(
+            organization, resource, {"quota_limited_until": billing_period_end, "quota_limiting_suspended_until": None}
+        )
+        return {
+            "quota_limited_until": billing_period_end,
+            "quota_limiting_suspended_until": None,
+        }
```

Comment:
I'm not sure if we want this here, as it will impact reporting: e.g., it will always report as `suspended` and never reach `already limited`.

So I'd either
- a) Define a const with quota resources that should skip `never_drop_data` and trust scores, and use it to skip the if blocks related to those (it would then fall back to the last `else`, which currently isn't supposed to be reached, at least in theory)
- b) Move this new block after `already limited` (2a) and data retention feature flag (1c)

Also the numbers/letters were already a bit messed up - would be awesome to clean it up (including the main comment on lines 213-223) to reflect the new end state.

Reply:
Good point, my understanding is fairly shallow here. I simplified this by implementing `TRUST_EXEMPT_RESOURCES` and a negative trust score for AI credits, instead of this new `if` branch. Take a look! Any notes?

Reply:
Sounds good. Made a couple of tweaks (while resolving merge conflicts)
- Renamed to `GRACE_PERIOD_EXEMPT_RESOURCES`
- Switched from the negative trust score to just checking if a resource is among those excluded from grace period - felt better semantically, since exemption is not just based on trust scores but also never drop data

Pushed the latter as the last commit - hope you don't mind!